Sources/TuistGraph/Models/Arguments.swift

```diff
@@ -19,3 +19,32 @@ public struct Arguments: Equatable, Codable {
         self.launchArguments = launchArguments
     }
 }
+
+extension Arguments: Equatable {
+    /// Implement `Equatable` manually so order of arguments doesn't matter.
+    public static func == (lhs: Arguments, rhs: Arguments) -> Bool {
+        lhs.environment == rhs.environment
+            && lhs.launchArguments.sorted { $0.name < $1.name }
+            == rhs.launchArguments.sorted { $0.name == $1.name }
+    }
+}
```

Comment:
Let's move it where it's actually needed (i.e. in the tests)

Reply:
This is actually not possible, if the struct already implements `Equatable` you cannot override the `==` operator in an extension, here is an explanation from the swift forums https://forums.swift.org/t/implement-equatable-protocol-in-a-class-hierarchy/13844.
I tried doing it and as expected it didn't work.

Reply:
Given the argument order is not relevant, I think it's fine to have this as Argument Equatable